prusti-server/src/process_verification.rs

```diff
@@ -14,9 +14,88 @@ use prusti_common::{
 };
 use std::{fs::create_dir_all, path::PathBuf};
 use viper::{
-    smt_manager::SmtManager, Cache, VerificationBackend, VerificationContext, VerificationResult,
+    native_backend::native_verifier::NativeVerifier, smt_manager::SmtManager, Cache,
+    VerificationBackend, VerificationContext, VerificationResult, Verifier,
 };
 
+pub fn process_native_verification_request(
+    mut request: VerificationRequest,
+    cache: impl Cache,
+) -> viper::VerificationResult {
+    // Only for testing: Check that the normalization is reversible.
+    if config::print_hash() {
+        debug_assert!({
+            let mut program = request.program.clone();
+            let normalization_info = NormalizationInfo::normalize_program(&mut program);
+            normalization_info.denormalize_program(&mut program);
+            program == request.program
+        });
+    }
+
+    // Normalize the request before reaching the cache.
+    let normalization_info = NormalizationInfo::normalize_program(&mut request.program);
+
+    let hash = request.get_hash();
+    info!(
+        "Verification request hash: {} - for program {}",
+        hash,
+        request.program.get_name()
+    );
+
+    // Only for testing: Print the hash and skip verification.
+    if config::print_hash() {
+        println!(
+            "Received verification request for: {}",
+            request.program.get_name()
+        );
+        println!("Hash of the request is: {}", hash);
+        // TODO: Some tests need the dump to report a diff of the Viper programs.
+        return viper::VerificationResult::Success;
+    }
+
+    // Early return in case of cache hit
+    if config::enable_cache() {
+        if let Some(mut result) = cache.get(hash) {
+            info!(
+                "Using cached result {:?} for program {}",
+                &result,
+                request.program.get_name()
+            );
+            normalization_info.denormalize_result(&mut result);
+            return result;
+        }
+    };
+
+    assert!(request.backend_config.backend == VerificationBackend::Lithium);
+
+    // Create a new verifier each time.
+    // A workaround for https://github.com/viperproject/prusti-dev/issues/744
+    // TODO: Is it still needed?
+    let mut stopwatch = Stopwatch::start("prusti-server", "verifier startup");
```

Comment:
`prusti-server` is used for accommodating the JVM startup time and should not be needed with the native backend.

Reply:
Yes, I intend to purge all my changes from `prusti-server` eventually, but for now I just plugged this in to make the compiler happy.

Reply:
Do you mean that the message is misleading? The server does provide caching and a uniform API to verify programs.

Reply:
`prusti-server` is JVM dependent. If we want to use it for caching/network-based verification, I think we should create a server trait that is implemented by Viper and native servers.

Reply:
Good point, the JVM dependency is bad.